src/Core/Infrastructure/RealTime/Repository/Icon/DbIconFactory.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * Copyright 2005 - 2021 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Core\Infrastructure\RealTime\Repository\Icon;
+
+use Core\Domain\RealTime\Model\Icon;
+
+class DbIconFactory
+{
+    /**
+     * @param array<string, string|null> $data
+     * @return Icon|null
+     */
+    public static function createFromRecord(array $data): ?Icon
+    {
+        if (
+            $data['icon_name'] === null
+            && $data['icon_url'] === null
```

Comment:
```suggestion
            || $data['icon_url'] === null
```

Reply:
One or the other can be null,but not the two at the same time hence the &&

Reply:
Can we have an icon with a name but without url ?

Reply:
Regarding the database model I guess yes. Does it make sense ? I dont think so